addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2560,16 +2560,19 @@ const Many2oneUserValueWidget = SelectUserValueWidget.extend({
     async _search(needle) {
         this._userValueWidgets = this._userValueWidgets.filter(widget => !widget.isDestroyed());
         // Remove select options
-        this._userValueWidgets
-            .filter(widget => {
+        const clearSuggestions = () => {
+            this._userValueWidgets = this._userValueWidgets.filter(widget => !widget.isDestroyed());
```

Comment:
- Do we still need the first line if you duplicate the line inside here?
- Is this actually necessary in the first place? Don't we have a system that does that automatically on all widgets destroy?

Reply:
It is actually needed because of the `button.destroy()` that follows which does not clean up the list. :face_with_spiral_eyes:
I moved the cleanup after.
There is no general "magic" cleanup of that list whenever a destroy happens - but the whole list is flushed on specific case of  `_rerenderXML` after all its content is `destroy`ed.